File: parth/ContainsDuplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> dups;
        for(int& n : nums){
            if(!dups.insert(n).second){
                return true;
            }
        }
        return false;
    }
};

File: Manali/ArraysAndHashing/ContainsDuplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        return len(nums)!=len(set(nums))
    
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        unique = set()
        for num in nums:
            if num in unique: return True
            unique.add(num)
        return False
    
TC: O(n)
SC: O(n)

#also do by sorting O(nlogn) and space O(1)